src/components/ui/SlideOver/SlideOver.tsx

```diff
@@ -18,74 +18,34 @@ interface SlideOverProps extends HTMLAttributes<HTMLDivElement> {
    * the modal content
    */
   onDismiss?: () => void
-  /**
-   * This callback function enables the transition effect on children "close" button
-   */
-  onDismissTransition: (callback: () => unknown) => unknown
 }
 
 const SlideOver = ({
   isOpen,
   onDismiss,
-  onDismissTransition,
   direction = 'leftSide',
   size = 'full',
   children,
   ...otherProps
 }: SlideOverProps) => {
-  const [fadeType, setFadeType] = useState<FadeType>()
-  const layout = useRef<HTMLElement | null>(null)
-
-  const handleClose = useCallback(() => {
-    setFadeType('out')
-    if (layout.current) {
-      layout.current.classList.remove('no-scroll')
-    }
-  }, [layout])
-
-  useEffect(() => {
-    layout.current = document.getElementById('layout')
-  }, [])
+  const { fade, onModalOpen, onModalClose } = useModal()
 
   useEffect(() => {
-    if (isOpen) {
-      setFadeType('in')
-
-      // Avoids double scroll issue on the page
-      if (layout.current) {
-        layout.current.classList.add('no-scroll')
-      }
-    }
-
-    return () => {
-      if (layout.current) {
-        layout.current.classList.remove('no-scroll')
-      }
-    }
-  }, [isOpen, layout])
-
-  useEffect(() => {
-    onDismissTransition(() => handleClose())
-  }, [handleClose, onDismissTransition])
-
-  const handleTransitionEnd = () => {
-    if (fadeType === 'out') {
-      onDismiss?.()
-    }
-  }
+    isOpen && onModalOpen()
+  }, [isOpen, onModalOpen])
 
   return (
     <UIModal
       isOpen={isOpen}
       onDismiss={(e) => {
         e.preventDefault()
-        handleClose()
+        onModalClose()
       }}
       data-slide-over
       data-slide-over-direction={direction}
       data-slide-over-size={size}
-      data-slide-over-state={fadeType}
-      onTransitionEnd={handleTransitionEnd}
+      data-slide-over-state={fade}
+      onTransitionEnd={() => fade === 'out' && onDismiss?.()}
```

Comment:
is it possible to run into a race condition where `fade` is `in` and the `onDismiss` is not called, leaving the modal open?

Reply:
I believe it is not, we are safe against the race condition by only calling `onDismiss` after transition/animation ends and, before that, checking for the right `fade` value